src/lib/Name.hs

```diff
@@ -1724,6 +1860,15 @@ instance (Traversable f, SubstE v e) => SubstE v (ComposeE f e) where
 instance (Traversable f, Eq (f ()), AlphaEq e) => AlphaEqE (ComposeE f e) where
   alphaEqE (ComposeE xs) (ComposeE ys) = alphaEqTraversable xs ys
 
+instance (Traversable f, Hashable (f ()), AlphaHashableE e)
+         => AlphaHashableE (ComposeE f e) where
+  hashWithSaltE env salt (ComposeE xs) = do
+    let h = hashWithSalt salt $ void xs
+    flip execState h $
+      forM_ xs \x -> do
+        curHash <- get
+        put $ hashWithSaltE env curHash x
```

Comment:
Wouldn't it be sufficient to have `Foldable f` and then express this with a `fold`? That would feel more natural than going for the state effect IMO

Reply:
It's easy to mess these instances up so I was trying to be extremely explicit. But now that it's done, I can replace it with the one-liner.